code/weapon/beam.cpp

```diff
@@ -2214,7 +2218,13 @@ void beam_aim(beam *b)
 		if(b->target_subsys != NULL){
 			vm_vec_unrotate(&b->last_shot, &b->target_subsys->system_info->pnt, &b->target->orient);
 			vm_vec_add2(&b->last_shot, &b->target->pos);
-			vm_vec_sub(&temp, &b->last_shot, &b->last_start);
+
+			if ((b->subsys != NULL) && (b->subsys->system_info->flags2 & MSS_FLAG2_SHARE_FIRE_DIRECTION)) {
```

Comment:
nitpick; since we've moved to c++11 can you please replace NULL with nullptr? :)